modules/redcap/php/redcaphttpclient.class.inc

```diff
@@ -0,0 +1,761 @@
+        if (empty($event)) {
+            throw new \LorisException("[redcap] Error: required 'event'.");
+        }
+        if (empty($recordId)) {
+            throw new \LorisException("[redcap] Error: required 'recordId'.");
+        }
+
+        // mapping check
+        if (!$this->hasMappingInstrumentEvent($instrument, $event)) {
+            throw new \LorisException(
+                "[redcap] Error: mapping '$instrument'/"
+                . "'$event' does not exist in REDCap"
+            );
+        }
+
+        // request
+        $r = $this->_getRecords(
+            [$instrument],
+            [$event],
+            [$recordId]
+        );
+
+        if (empty($r)) {
+            throw new \LorisException("[redcap] Error: no data found.");
+        }
+
+        // Only keep complete records
+        if ($completedRecordsOnly) {
+            $completed = array_filter(
+                $r,
+                function ($record) use ($instrument) {
+                    return $record[$instrument . "_complete"] == 2;
+                }
+            );
+
+            if (count($completed) < 1) {
+                throw new \LorisException(
+                    "[redcap] Error: no complete record found."
+                );
+            }
+        } else {
+            // if not only completed records
+            $completed = $r;
+        }
+
+        // Order the records by ${instrument_name}_dtt field value
+        usort(
+            $completed,
+            function ($a, $b) use ($instrument) {
+                $dttField = $instrument . "_dtt";
+                $a_date   = new \DateTimeImmutable($a[$dttField]);
+                $b_date   = new \DateTimeImmutable($b[$dttField]);
+                return $a_date <=> $b_date;
+            }
+        );
+
+        // is a repeating instrument?
+        $final = [];
+        if ($this->getProjectInfo()->hasRepeatingInstrumentsOrEvents()
+            && $this->hasRepeatingInstrumentEvent($instrument, $event)
+        ) {
+            foreach ($completed as $index => $record) {
+                $final[] = new RedcapRepeatingRecord(
+                    $instrument,
+                    $record,
+                    $index + 1
+                );
+            }
+        } else {
+            // return the only record
+            $final[] = new RedcapRecord($instrument, $completed[0]);
+        }
+        return $final;
+    }
+
+    /**
+     * Get records from REDCap.
+     * Cannot return all values. At least one of the three parameters needs to
+     * be filled with one value, else an exception will be generated.
+     *
+     * @param array $instruments instrument names
+     * @param array $events      event names
+     * @param array $records     records IDs
+     *
+     * @return array an array of records (array of [field name => field values])
+     */
+    private function _getRecords(
+        array $instruments = [],
+        array $events = [],
+        array $records = []
```

Comment:
I would also use more descriptive names here like `$instrumentNames`, `$eventNames` and `$recordIds`